data/txHandler.go

```diff
@@ -328,19 +414,134 @@ type SolicitedTxHandler interface {
 	Handle(txgroup []transactions.SignedTxn) error
 }
 
+type solicitedTxHandler struct {
+	txHandler *TxHandler
+}
+
 // SolicitedTxHandler converts a transaction handler to a SolicitedTxHandler
 func (handler *TxHandler) SolicitedTxHandler() SolicitedTxHandler {
 	return &solicitedTxHandler{txHandler: handler}
 }
 
-type solicitedTxHandler struct {
-	txHandler *TxHandler
-}
-
 func (handler *solicitedTxHandler) Handle(txgroup []transactions.SignedTxn) error {
-	outmsg, _ := handler.txHandler.processDecoded(txgroup)
-	if outmsg.Action == network.Disconnect {
+	disconnect := handler.txHandler.processDecoded(txgroup)
+	if disconnect {
 		return fmt.Errorf("invalid transaction")
 	}
 	return nil
 }
+
+// SolicitedAsyncTxHandler handles messages received through channels other than the gossip network.
+// It therefore circumvents the notion of incoming/outgoing messages
+type SolicitedAsyncTxHandler interface {
```

Comment:
generally speaking - yes. In this case, I think that it might be redundant, but not really harmful. It's usually a great way to perform testing. Most of this file would probably need to be rewritten once we eliminate the old transaction handler code.